who-metrics-ui/src/components/RepositoriesTable.tsx

```diff
@@ -104,7 +117,138 @@ const MinMaxRenderer: FC<{
               onClick={(e) => e.stopPropagation()}
             />
           </FormControl>
-        </div>
+        </Box>
+      )}
+    </HeaderCellRenderer>
+  );
+};
+
+// Renderer for the searchable select filter
+const SearchableSelectRenderer: FC<{
+  headerCellProps: RenderHeaderCellProps<Repo>;
+  filters: Filter;
+  updateFilters: ((filters: Filter) => void) &
+  ((filters: (filters: Filter) => Filter) => void);
+  filterName: keyof Filter;
+}> = ({ headerCellProps, filters, updateFilters, filterName }) => {
+  const [filteredOptions, setFilteredOptions] = useState<string>('');
+  const allSelectOptions = dropdownOptions(filterName, filteredOptions);
+
+  return (
+    <HeaderCellRenderer<Repo> {...headerCellProps}>
+      {({ ...rest }) => (
+        <Box>
+          <TextInput
+            {...rest}
+            className="w-full"
+            value={filteredOptions}
+            onChange={(e) => setFilteredOptions(e.target.value)}
+            trailingAction={
+              <TextInput.Action
+                onClick={() => {
+                  setFilteredOptions('');
+                }}
+                icon={XIcon}
+                aria-label="Clear input"
+                sx={{ color: 'fg.subtle' }}
+              />
+            }
+          />
+          <Box className="h-80 overflow-auto w-100 mt-2">
+            <ActionList >
+              <ActionList.Item
+                onClick={() => {
+                  updateFilters((otherFilters) => ({
+                    ...otherFilters,
+                    [filterName]: {
+                      ...otherFilters[filterName],
+                      all: !(
+                        otherFilters[filterName] as Record<string, boolean>
```

Comment:
The case here is because our filters allow both string and number maps

```
type Filter = {
  repositoryName?: Record<string, boolean>;
  licenseName?: Record<string, boolean>;
  collaboratorsCount?: Array<number | undefined>;
  watchersCount?: Array<number | undefined>;
  openIssuesCount?: Array<number | undefined>;
  openPullRequestsCount?: Array<number | undefined>;
  closedIssuesCount?: Array<number | undefined>;
  mergedPullRequestsCount?: Array<number | undefined>;
  forksCount?: Array<number | undefined>;
};
```

Compiler doesn't know which one it will be (but we know it's not the number array)

Reply:
Ah gotcha! That makes sense. I feel like there might be another way to work around it, but I don't have any great ideas off the top of my head, so happy to live with the cast for now 🙃